organization/src/main/scala/com/improving/app/organization/domain/Organization.scala

```diff
@@ -1,6 +1,206 @@
 package com.improving.app.organization.domain
 
-object Organization{
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+import akka.pattern.StatusReply
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.scaladsl.{Effect, EventSourcedBehavior, ReplyEffect}
+import com.google.protobuf.timestamp.Timestamp
+import com.improving.app.common.domain.MemberId
+import com.improving.app.common.errors.Validation._
+import com.improving.app.common.errors._
+import com.improving.app.organization.domain.OrganizationValidation._
 
+import java.time.Instant
+
+object Organization {
+  val TypeKey = EntityTypeKey[OrganizationRequestEnvelope]("Organization")
+
+  case class OrganizationRequestEnvelope(request: OrganizationRequestPB, replyTo: ActorRef[StatusReply[OrganizationEvent]])
+
+  sealed trait OrganizationState
+
+  private case object UninitializedState extends OrganizationState
+
+  private sealed trait EstablishedState extends OrganizationState {
+    val info: OrganizationInfo
+    val metaInfo: OrganizationMetaInfo
+  }
+  private case class DraftState(info: OrganizationInfo, metaInfo: OrganizationMetaInfo) extends EstablishedState
+
+  private case class ActiveState(info: OrganizationInfo, metaInfo: OrganizationMetaInfo) extends EstablishedState
+
+  private case class SuspendedState(info: OrganizationInfo, metaInfo: OrganizationMetaInfo) extends EstablishedState
+
+  def apply(persistenceId: PersistenceId): Behavior[OrganizationRequestEnvelope] = {
+    Behaviors.setup(context =>
+      EventSourcedBehavior[OrganizationRequestEnvelope, OrganizationEvent, OrganizationState](
+        persistenceId = persistenceId,
+        emptyState = UninitializedState,
+        commandHandler = commandHandler,
+        eventHandler = eventHandler
+      )
+    )
+  }
+
+  private val commandHandler: (OrganizationState, OrganizationRequestEnvelope) => ReplyEffect[OrganizationEvent, OrganizationState] = { (state, command) =>
+    val result: Either[Error, OrganizationEvent] = state match {
+      case UninitializedState =>
+        command.request.asMessage.sealedValue match {
```

Comment:
I think if you just make this line `command.request`, you can change the rest of the cases to look something like `x: EstablishOrganization => ...`

Reply:
That's a lot nicer. Thanks!